### PR TITLE
Full-width admin layout

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

Replaces the static 960px grid in admin layout with a fluid full-width grid. :tada:

#### What should we test?

All pages in admin look okay visually.

#### Release notes

Improved admin layout

#### Screenshot

Screenshot from 2020-09-21 15-32-22

### CODE REVIEW

**Comment by RachL** [comment]
@Matt-Yorkley any chance you can update us of where you are here? It looks like a big change :)

**Comment by Matt-Yorkley** [comment]
@RachL this was just a quick experiment I did on a Sunday hack-session, not sure if we want to use it or not?

**Comment by luisramos0** [comment]
I gave this one a try locally. It works :-O
There are some very minor fixes we would need and maybe adapt some of the pages to look better in large screens.
These are the only problems I found:

Large labels dont fit placeholder:
image
Menu in 2 lines:
image
Form fields not aligned:
image

I'd recommend someone from Product or Design to stage this and have a look. It's awesome!

**Comment by Matt-Yorkley** [comment]
Thanks @luisramos0, I did a quick pass already fixing a few places where the existing CSS/HTML wasn't designed to work outside of the 960 grid, but there's obviously a couple of bits I missed :+1: 

Also I only tested it on my medium screen and hadn't tried it on a large screen yet.

**Comment by Matt-Yorkley** [comment]
Okay, I think I've fixed the menu on two lines issue (and deleted two javascript dependencies in the process :tada:)

**Comment by lin-d-hop** [comment]
@Matt-Yorkley this is remarkable!
This would be a huge win to push through.
I only noticed the issues that Luis has already pointed out (will redeploy once the build finishes). I haven't done a systematic sweep but I've had a decent look. I think this warrants the little product and design time needed to move through the pipe.

File: app/assets/stylesheets/admin/all.scss
Comment (luisramos0) [inline app/assets/stylesheets/admin/all.scss:8]: this can be removed from vendor as well?

File: app/assets/stylesheets/admin/grid.scss
Comment (luisramos0) [inline app/assets/stylesheets/admin/grid.scss:1]: extra empty line at the top.

File: app/views/admin/enterprises/form/_shop_preferences.html.haml
Comment (luisramos0) [inline app/views/admin/enterprises/form/_shop_preferences.html.haml:19]: this is a missing translation.

**Review by luisramos0 — APPROVED** [review]
from a code perspective, this is good  :+1: 
we need to test it.

**Comment by luisramos0** [comment]
any reason for it to be a draft Matt?

File: app/assets/stylesheets/admin/all.scss
Comment (Matt-Yorkley) [reply app/assets/stylesheets/admin/all.scss:8]: This is actually in `spree_core`. Maybe I need to rebase and have another look...

Comment (Matt-Yorkley) [reply app/assets/stylesheets/admin/all.scss:8]: Done.

File: app/views/admin/enterprises/form/_shop_preferences.html.haml
Comment (Matt-Yorkley) [reply app/views/admin/enterprises/form/_shop_preferences.html.haml:19]: Done here: https://github.com/openfoodfoundation/openfoodnetwork/pull/6281

File: app/assets/stylesheets/admin/all.scss
Comment (luisramos0) [reply app/assets/stylesheets/admin/all.scss:8]: yes, I brought it from spree recently :+1:

File: app/views/admin/enterprises/form/_shop_preferences.html.haml
Comment (luisramos0) [reply app/views/admin/enterprises/form/_shop_preferences.html.haml:19]: awesome!

**Review by luisramos0 — APPROVED** [review]

**Review by mkllnk — APPROVED** [review]
Ingenious! :crown: 

I'm looking forward to seeing that in production. :grinning:

**Comment by mkllnk** [comment]
@Matt-Yorkley This needs rebasing.

**Comment by Matt-Yorkley** [comment]
Okay, this is rebased now. It will need some careful checking for visual regressions. There might be some elements that look slightly the wrong size or out of position after this change. It should be easy to adjust them but we need to have a look and find them.

**Comment by filipefurtad0** [comment]
Hi @Matt-Yorkley ,

looks like a huge improvement... plenty of space now!

As you mention, I think having all this space in cells makes some formatting issues easier to spot. I went through all or most admin pages and would mention some cases which could perhaps be improved.

**Possible improvements**

- Tables: text centering / space management inside cells

Through out the app one can spot that the text is centered in some cells, not being the case in others, for example:

Orders page
image

Inventory page
image

Customers
image


Shipping and payment methods (from dashboard) - maybe these could be centered?

image

image


- date pickers and enterprise permissions

In these examples I think the empty space could be perhaps used, to give some more room to date pickers:

image

Same case here for enterprise permissions:

image


**Broken or unusable pages**

- enterprise fees page seems a bit broken now:

image

- adding a new payment method or shipping method - I think this is the only actual regressions, which makes the pages not usable:

image

image


Other than the issues above, perhaps others would like to have a look as well, maybe @Erioldoesdesign and/or @jaycmb ? I'll leave this PR staged in staging-FR (https://staging.coopcircuits.fr/).

PS: still to-do, test superadmin pages.

**Comment by Matt-Yorkley** [comment]
I added several commits to improve those highlighted bits :+1:

**Comment by jaycmb** [comment]
This is awesome!! So much space :)

We will discuss the scope of design work that could be done alongside this release in product curation meeting.
Issue created here https://app.zenhub.com/workspaces/product--inception-pipe-5f82e534db531f000f8c09ff/issues/openfoodfoundation/inception-pipe/24

**Comment by filipefurtad0** [comment]
Hey @Matt-Yorkley ,

Thanks for all the work on this. I checked the previously broken pages:

- enterprise fees
image


- creating a payment method
image

- creating a shipping method
image

These are now fixed.

It's awesome to be able to use full-width to display all the columns correctly - for example, in Bulk Order Management :tada: 
image


Also, I checked superadmin pages and found everything to work. The admin pages are not exceptionally usable on mobile, but this is unrelated to this PR. So, from the functional side of things I think this would be good to go :+1: 

The PR will remain on test-ready awaiting feedback from Product, as previously discussed.

**Comment by sigmundpetersen** [comment]
We should probably release the staging server on this while waiting for inception/feedback?

**Comment by Matt-Yorkley** [comment]
I'm stealing this staging server for a bit, feel free to re-stage this when needed :+1:

**Comment by Erioldoesdesign** [comment]
Hey folks! here's the design sweep/QA of this: https://docs.google.com/presentation/d/1D0nqGRMhPaadgVd7KgS99gq2WqSvjgbKhaUKd4CZeYk/edit#slide=id.gb1b5274d1d_0_660

we'll be chatting out it in product curation today 11.01.2021

**Comment by RachL** [comment]
@Matt-Yorkley we can merge this one :muscle: it's easier if @jibees works on improvements on master and it's not introducing severe regression.

Anything we need to update before merging? (I think this PR was created before the last rudy upgrade, but maybe no action needed?)

**Comment by Matt-Yorkley** [comment]
I did a quick rebase. Seems like none of these bits of code have been touched since it was submitted :+1:

**Pull request merged**
